src/presentation/http/router/note.test.ts

```diff
@@ -49,4 +52,83 @@ describe('Note API', () => {
       expect(body).toStrictEqual({ message: 'Note not found' });
     });
   });
-});
+
+  describe('GET note/:notePublicId ', () => {
+    test('Returns note by public id with 200 status when note is publicly available', async () => {
+      const expectedStatus = 200;
+      const correctID = 'Pq1T9vc23Q';
+
+      /**
+       * @todo API should not return internal id and "publicId".
+       * It should return only "id" which is public id. Not implemented yet.
+       */
+
+      const expectedNote = {
+        'id': 2,
+        'publicId': 'Pq1T9vc23Q',
+        'creatorId': 1,
+        'content': null,
+        'createdAt': '2023-10-16T13:49:19.000Z',
+        'updatedAt': '2023-10-16T13:49:19.000Z',
+      };
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual(expectedNote);
+    });
+
+    test('Returns 403 when public access is disabled, user is not creator of the note', async () => {
+      const expectedStatus = 403;
+
+      const notPublicNote = notes.find(newNote => {
+        const settings = noteSettings.find(ns => ns.note_id === newNote.id);
+
+        return settings!.is_public === false;
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${notPublicNote!.public_id}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Permission denied' });
+    });
+
+    test('Returns 406 when the id  does not exist', async () => {
+      const expectedStatus = 406;
+      const nonexistentId = 'ishvm5qH84';
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${nonexistentId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
+
+    test('Returns 406 when the id contains incorrect characters', async () => {
```

Comment:
"Incorrect" id is not only about characters, but also length. If you'll pass "aaaa" — characters will be correct, but length is not


```suggestion
    test('Returns 406 when passed id is incorrect', async () => {
```